patches/tModLoader/Terraria/ModLoader/Core/ModOrganizer.cs

```diff
@@ -136,6 +136,7 @@ void FilterOut(Func<LocalMod, bool> condition, string reason)
 		OrderByDescending(m => m.tModLoaderVersion, "a matching version for a newer tModLoader exists");
 		FilterOut(m => m.location != ModLocation.Workshop && list.Any(m2 => m2.location == ModLocation.Workshop && m2.modFile.Hash == m.modFile.Hash), "an identical copy exists in the workshop folder");
 		OrderByDescending(m => m.location == ModLocation.Local, "a local copy with the same version (but different hash) exists");
+		FilterOut(m => m.location == ModLocation.Local && Path.GetFileNameWithoutExtension(m.modFile.path) != m.Name && list.Any(m2 => m2.location == ModLocation.Local && Path.GetFileNameWithoutExtension(m2.modFile.path) == m.Name), "this is a renamed duplicate of another .tmod file in the local mods folder");
```

Comment:
There are a couple of ways to make this simpler. Note that at this point:
- All mods have the same `version` and `tModLoaderVersion`
- All mods are in the same `ModLocation`
- The mods may have different hashes, in which case we select one arbitrarily. Thus it's not 100% accurate that they're all 'duplicates'

You want to prefer mods with a 'canonical' name. There's no need to check if there is another mod with a canonical name. You can use either `FilterOut` or `OrderByDescending`, the only difference is that if you use `FilterOut` and there are _no_ mods with a canonical name, then a warning will be logged.

I recommend:
`FilterOut(m => Path.GetFileNameWithoutExtension(m.modFile.path) != m.Name, "The file name and internal mod name do not match")`

If you don't want the warning:
`OrderByDescending(m => Path.GetFileNameWithoutExtension(m.modFile.path) == m.Name, "this .tmod has been renamed")`

It should be fine to apply this canonical naming preference/condition to mods from non-local locations, this there's no need for the `m.location == ModLocation.Local` condition.

Reply:
Sounds good, I went with OrderByDecending because I imagine there are some who end up with a renamed tmod file as their only tmod file for a specific mod and showing an error to them doesn't seem useful.